crates/keystore/src/lib.rs

```diff
@@ -33,6 +39,87 @@ pub struct KeyStore<'a> {
     filename: String,
     keys: Vec<Jwk>,
 }
+use chacha20poly1305::{
+    aead::{Aead, AeadCore, OsRng},
+    ChaCha20Poly1305,
+};
+
+use log::{debug, info, error}; // Import logging macros
+use secrecy::{ExposeSecret, SecretString};
+use zeroize::Zeroize;
+
+// Define a custom error type for keystore operations
+#[derive(Debug, thiserror::Error)]
+pub enum KeystoreError {
+    #[error("File error: {0}")]
+    FileError(std::io::Error),
+    #[error("Encryption error: {0}")]
+    EncryptionError(chacha20poly1305::Error),
+    #[error("Decryption error: {0}")]
+    DecryptionError(chacha20poly1305::Error),
+}
```

Comment:
Why not use the existing keystore error enum, and also we have to many errors now it will be better to place the errors in a different file